### DISCUSSION TITLE
Resolving `numpy.typing` import

### DISCUSSION THREAD

**Discussion by Tseyang** [microsoft/pyright#3264] [Q&A] [answered]
I've recently picked up using pyright and have been loving its integration with VSCode so far. I'm running into an issue with `numpy.typing` where it can't seem to detect the import:

<img width="463" alt="image" src="https://user-images.githubusercontent.com/20376539/160429321-7045c3e4-0137-482a-9d42-a77aba243a93.png">

For context, the file has:
```python3
import logging
import numpy as np
import numpy.typing as npt
from numpy import float64
from pandas import DataFrame
```

I know I'm definitely doing something wrong since I've seen multiple examples of folks importing `numpy.typing` without issue. But I'm not sure how I should be debugging this. So far what I've checked:

- I am using the right Python Interpreter for my LSP (Poetry shell, python 3.10)
- My `numpy` version is 1.22.3, which should have the typing module FWICT
- my `pyproject.toml` has 

```toml
[tool.poetry.dependencies]
python = "~3.10"
numpy = "^1.22.3"
```

I appreciate the help and this great tool!

**Comment by erictraut** [comment, marked as answer]
I'm not able to repro the problem as reported. Is it possible you have an old numpy stub package installed? If you right click on the word "typing" in the statement `import numpy.typing as npt` and choose "Go To Declaration", where does it take you? It should take you to `site-packages/numpy/typing/__init__.py`.

Here are a few other things you can try to narrow down the problem.
1. Try to repro with a fresh venv with no packages installed other than numpy.
2. Create a pyrightconfig.json file in your project root and add `{ "verboseOutput": true }` to it. This will cause additional import resolution details to be logged to the "Output" tab in VS Code.

BTW, if you're using pyright with VS Code, we recommend that you use Pylance instead. It builds on pyright and adds a bunch of useful convenience features.

**Reply by Tseyang** [reply]
@erictraut Thank you for the fast response + good suggestion. I realized I had a stale `data-science-types` package that was not removed properly. I removed it and I think its working now!

**Reply by andrewkittredge** [reply]
I had to delete the numpy-stubs directory

**Reply by Gideon-Felt** [reply]
Yeah, I had to remove `data-science-types` too, resolved the issue.
